docs/architecture/graphql.rst

```diff
@@ -0,0 +1,109 @@
+GraphQL API (Beta)
+====================
+
+.. note::
+
+    The GraphQL API is in the early version and is subject to change.
+
+
+Saleor supports GraphQL API. It allows to create apps that can handle fulfillment, orders, pages, product and shipping management.
+
+GraphQL allows for more efficient data retrieval - it is easier to fetch all desired resources without redundant ones.
+
+Learn more at `official website <https://graphql.org>`_.
+
+Endpoint
+--------
+You can test API under ``/graphql`` endpoint - in default Saleor configuration URL would be ``http://localhost:8000/graphql/``. You can perform queries normally, however you need to log in to an account with proper permissions to run mutations and query for restricted data.
+
+
+Query
+-----
+Saleor GraphQL uses `Relay connection standard <facebook.github.io/relay/graphql/connections.htm>`_, which is very handy in case of pagination.
```

Comment:
I think pagination deserves its own paragraph. We can skip it for now and add it later I guess.